### PR TITLE
Add jgit health indicator

### PR DESCRIPTION
**Pull request by vasilievip**
Add HEAD commit info from jgit repository to /health endpoint

### CODE REVIEW

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/AbstractScmEnvironmentRepository.java [hunk 1]
```diff
@@ -44,8 +47,15 @@ public synchronized Environment findOne(String application, String profile, Stri
...
-		return this.cleaner.clean(result, getWorkingDirectory().toURI().toString(),
-				getUri());
+		StringBuilder state = new StringBuilder();
+		if(locations.getTimestamp() > 0) {
+			Date date = new Date(locations.getTimestamp());
+			state.append(DateFormat.getDateTimeInstance().format(date));
+			state.append(" ");
+		}
+		state.append(locations.getInfo());
+		result.setState(state.toString());
```
Comment (spencergibb) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/AbstractScmEnvironmentRepository.java]: The `Environment.state` was included for client side state, not server side.

Comment (vasilievip) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/AbstractScmEnvironmentRepository.java]: not sure I understood the problem. If Environment is client data structure, why then its used everywhere on server? If field state is client only, why there is no ClientEnvironment with this field and ServerEnvironment w/o this field. Should I add 1 or2 more fields to Environment to transfer this information? Why this field can't be used on server side?

Comment (spencergibb) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/AbstractScmEnvironmentRepository.java]: It's a representation of the client state.

Comment (vasilievip) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/AbstractScmEnvironmentRepository.java]: Added new field

**Pull request closed without merging**